### ISSUE TITLE
LitGPTSDPABenchmark runs incorrect configs

### ISSUE THREAD

**Issue by vedaanta** [Lightning-AI/lightning-thunder#317]
## 🐛 Bug
`LitGPTSDPABenchmark`, located [here](https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/benchmarks/__init__.py), does not correctly run the litgpt config provided to it.
This is due using `NanoGPTConfig` underneath, located [here](https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/benchmarks/__init__.py#L1157).

This creates multiple issues when benchmarking, as incorrect sdpa operation is launched. For example:
1. litgpt never has dropout ON. But as `NanoGPTConfig` has default dropout of `0.1`, all sdpa operations have dropout ON.
2. Running GQA sizes, with `config.n_query_groups`, it is not propagated into the model.

### To Reproduce
```
from thunder.benchmarks import LitGPTSDPABenchmark, Benchmark
import thunder
import torch

bench: Benchmark = LitGPTSDPABenchmark(
        config="Llama-2-7b-hf", device="cuda:0", dtype=thunder.bfloat16, requires_grad=True
    )

args, kwargs = bench.make_batch()
torch.cuda.synchronize()
fn = bench.fn()

jfn = thunder.jit(fn)
jfn(*args, **kwargs)

print(thunder.last_traces(jfn)[-1])
```

Outputs:
```
# Constructed by Delete Last Used (took 0 milliseconds)
import torch
from thunder.executors.torchex import no_autocast

@torch.no_grad()
@no_autocast
def augmented_forward_fn(q, k, v):
  # q: "cuda:0 bf16[16, 32, 4096, 128]"
  # k: "cuda:0 bf16[16, 32, 4096, 128]"
  # v: "cuda:0 bf16[16, 32, 4096, 128]"
  (t0, t1, t2, t3, _, _, t4, t5, _) = sdpafx_grad_forward_scaled_dot_product_efficient_attention(q, k, v, 0.1, True, scale=None)
  return {'output': t0, 'flat_args': [q, k, v], 'flat_output': (t0,)}, ((k, q, t0, t1, t2, t3, t4, t5, v), (True, 0.1))
```

Notice the `is_causal=True`, and `dropout_p=0.1`.
litgpt model has no code path to enable both of them. [code](https://github.com/Lightning-AI/litgpt/blob/main/litgpt/model.py#L247)

### Expected behavior
`LitGPTSDPABenchmark` runs the exact parameters specified by litgpt configs.

cc @crcrpar @carmocca

**Comment by mruberry** [comment]
triage review — @riccardofelluga, is this something you'd like to look at?

**Comment by carmocca** [comment]
> Running GQA sizes, with config.n_query_groups, it is not propagated into the model.

Note that litgpt expands this dimension during training for flash attention support https://github.com/Lightning-AI/litgpt/blob/main/litgpt/model.py#L226-L231
